nemo/core/classes/modelPT.py

```diff
@@ -1390,29 +1145,9 @@ def _set_model_guid(self):
 
             appstate.register_model_guid(self.model_guid, restoration_path=restore_path)
 
-    @staticmethod
-    def _is_restore_type_tarfile() -> bool:
-        """
-        Utility method that checks if the restore path of the underlying Model
-        is a tarfile (can be any valid archive)._MODEL_EFF_SAVE
-        """
-        app_state = AppState()
-
-        if app_state.model_restore_path is None:
-            return False
+    @classmethod
+    def update_save_restore_connector(cls, save_restore_connector):
+        if hasattr(cls, '_save_restore_connector'):
+            cls._save_restore_connector = save_restore_connector
```

Comment:
in line 113 (constructor) you are setting that as instance (self) attribute.

If you would set it as class attribute, in __init_subclass__ method, then this (and generally logic of handling cls._save_restore_connector) could be simplified...

Reply:
I'll look into this.

Reply:
I added this and we were able to remove the connector arg from register_artifact.